aws_lambda_powertools/tracing/provider/base.py

```diff
@@ -0,0 +1,85 @@
+import abc
+import numbers
+from contextlib import asynccontextmanager, contextmanager
+from typing import AsyncGenerator, Generator, Sequence, Union
+
+
+class BaseSpan(abc.ABC):
+    """Holds common properties and methods on span."""
+
+    @abc.abstractmethod
+    def set_attribute(self, key: str, value: Union[str, numbers.Number, bool], **kwargs) -> None:
+        """set attribute for span with a key-value pair.
+
+        Parameters
+        ----------
+        key: str
+            Attribute key
+        value: Union[str, numbers.Number, bool]
+            Attribute value
+        """
+
+    @abc.abstractmethod
+    def record_exception(self, exception: BaseException, **kwargs):
+        """Add an exception to trace entities.
```

Comment:
```suggestion
        """Records an exception.
```

I'm sorry but "to trace entities" means nothing in my mind :( Can it maybe be rephrased?